### PR TITLE
integration: Add docker containers integration-tests

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
Add support for running integration-tests for non-Kubernetes containers. The implementation is divided into following steps:
- Introduce a package [`container-utils/testutils`](https://github.com/inspektor-gadget/inspektor-gadget/pull/1162/commits/794b10e222c73b5f3a7ea86e5c8d368b1553ae94) to facilitate management of non-Kubernetes containers. Currently, only `docker` is supported. Use functional options to make creation API cleaner. 
- Introduce an TestStep [interface](https://github.com/inspektor-gadget/inspektor-gadget/pull/1162/commits/7889b2adbb2b78261d69e30bad5d72dbaaa7da81) allowing a  consistent test workflow (Courtesy @blanquicet )
-  Adds tests for docker containers and hook them in the CI.

### CODE REVIEW

File: integration/local-gadget/docker/list_containers_test.go
Comment (blanquicet) [inline integration/local-gadget/docker/list_containers_test.go:119]: I was wondering if we could reuse the logic we already have to run commands to run these together. What about using an interface like [this](https://github.com/inspektor-gadget/inspektor-gadget/commit/c6d69f16061f2fe0c12a3152b019f5f8e6e68bfe)? WDYT?

**Review by blanquicet — COMMENTED** [review]
Great to see this initial support! Thanks! I added some comments, but it looks great.

File: integration/local-gadget/docker/list_containers_test.go
Comment (mqasimsarfraz) [reply integration/local-gadget/docker/list_containers_test.go:119]: Great suggestion. If we could utilize the `RunCommands` it would keep writing tests simple. I see we are trying to introduce an interface in #1116 as well. We can coordinate both the changes and simply this. 

cc: @burak-ok

Comment (blanquicet) [reply integration/local-gadget/docker/list_containers_test.go:119]: I think #1116 will not be used because we didn't see a real advantage. So, you can go ahead here.

Comment (blanquicet) [reply integration/local-gadget/docker/list_containers_test.go:119]: In any case, in case we need to do something similar, the interface I'm proposing is generic enough to add any additional operation in the future.

Comment (mqasimsarfraz) [reply integration/local-gadget/docker/list_containers_test.go:119]: > In any case, in case we need to do something similar, the interface I'm proposing is generic enough to add any additional operation in the future.

+1, I can start with something minimal and can always be extended later on.

File: pkg/container-utils/testutils/options.go
Comment (blanquicet) [inline pkg/container-utils/testutils/options.go:83]: Do we really need these two functions? Maybe mix them into one or just define the one that sets the opposite to the default value?

File: integration/command.go
Comment (blanquicet) [inline integration/command.go:278]: Mmmm cc @alban?

**Review by blanquicet — APPROVED** [review]
I like it. Some comments, but it looks good to me.

File: integration/opeartions.go
Comment (mqasimsarfraz) [reply integration/opeartions.go:22]: True, naming is so hard :D I actually liked `Operation` but adding `Test` prefix (`TestOperation`) might be bit more clear.  Also fine with `TestStep`. What do you say?

Comment (blanquicet) [reply integration/opeartions.go:22]: If we add the `Test` prefix, I will go for `TestStep`.

File: integration/command.go
Comment (blanquicet) [reply integration/command.go]: But if I am running local-gadget to debug a Kubernetes cluster, the test will try to print the IG logs, isn't it? If so, I think you can think about another solution for this in another PR to avoid blocking this one.

**Review by blanquicet — APPROVED** [review]
IMO, it is ready to go. I let you the decision of the naming :)

File: integration/docker.go
Comment (blanquicet) [inline integration/docker.go:31]: It doesn't need to be exposed, isn't it?

```suggestion
	started      bool
```

File: integration/command.go
Comment (blanquicet) [reply integration/command.go]: I didn't test it, but do you think something like https://github.com/inspektor-gadget/inspektor-gadget/commit/52e7e35ee696289827637ef05076711b40b37d7e may work?

Comment (mqasimsarfraz) [reply integration/command.go]: This is looking like something we can work with. I will handle it like that!

**Comment by mqasimsarfraz** [comment]
> IMO, it is ready to go. I let you the decision of the naming :)

After bit more thinking I felt that `TestStep` seems better so I updated the PR. :) Can you please have a look again before I merge this?

File: integration/docker.go
Comment (blanquicet) [inline integration/docker.go:24]:
```suggestion
// DockerContainer implements TestStep for docker containers
```

**Review by blanquicet — APPROVED** [review]
LGTM! Thanks for managing all comments!

**Pull request merged**
